Biz/Model/UserLoginBaseForm.pm

```diff
@@ -113,6 +125,48 @@ sub validate {
     return;
 }
 
+sub validate_and_execute_ok {
+    my($self) = @_;
+    my($res) = shift->SUPER::validate_and_execute_ok(@_);
+    # Won't have a realm owner if the login field was invalid
+    return $res
+        unless my $owner = $self->unsafe_get('realm_owner');
+    my($errors) = $self->get_errors || {};
+    if ($errors->{'login'} && $errors->{'login'}->eq_locked_out) {
+        b_warn('locked out owner=', $owner);
+        # Return non-specific errror so potential attacker doesn't glean additional information
+        return {
+            method => 'server_redirect',
+            task_id => 'DEFAULT_ERROR_REDIRECT',
+            query => undef,
+        };
+    }
+    # Have to do this after form processing as the transaction gets rolled back upon form errors
+    if (
+        $errors->{'RealmOwner.password'}
+        && _record_login_attempt($self, $owner, 0)->get('login_attempt_state')->eq_lockout
+    ) {
+        b_warn('failed login attempt triggered lock out owner=', $owner);
+        $self->req->with_realm($owner, sub {
+            $self->internal_put_field(
+                'Email.email' => $self->new_other('Email')->load({
```

Comment:
In this situation, keep this code very simple to avoid errors that might cause a rollback. The lockout mail task can then get `$owner` from this form and do the step of looking up an email. The mail task can then render the page that says the account is locked.